### PR TITLE
Adjust usage

### PR DESCRIPTION
**Pull request by JoviDeCroock**

### CODE REVIEW

File: packages/graphqlsp/src/ast/resolve.ts [hunk 1]
```diff
@@ -34,37 +36,33 @@ export function resolveTemplate(
...
-        if (!node || !node.parent) return;
+        const declaration = getDeclarationOfIdentifier(
+          span.expression,
+          typeChecker
+        );
+        if (!declaration) return;
 
-        const parent = node.parent;
+        const parent = declaration;
         if (ts.isVariableDeclaration(parent)) {
```
Comment (kitten) [inline packages/graphqlsp/src/ast/resolve.ts:49]: If I'm not mistaken, this is now redundant with `getValueOfValueDeclaration`. Unless we have extra logic for the discovered node (which shouldn't be called `parent` any longer but just `declaration` / `valueDeclaration` here, we can just use `getValueOfValueDeclaration` and do further operations separately, basically on:
- valueDeclarationNode
- valueNode

As long as tests pass, we can adjust this manually in a future PR
This change will cover more edge cases though once adjusted as per the above

File: packages/graphqlsp/src/ast/resolve.ts [hunk 2]
```diff
@@ -34,37 +36,33 @@ export function resolveTemplate(
...
     .map(span => {
       if (ts.isIdentifier(span.expression)) {
-        const definitions = info.languageService.getDefinitionAtPosition(
-          filename,
-          span.expression.getStart()
-        );
-
-        const def = definitions && definitions[0];
-        if (!def) return;
+        const typeChecker = info.languageService.getProgram()?.getTypeChecker();
```
Comment (kitten) [inline packages/graphqlsp/src/ast/resolve.ts:39]: This is awkwardly repeated now across the codebase. We can manually refactor our helper functions to just accept the `typeChecker` directly now afaict

File: packages/graphqlsp/src/ast/index.ts [hunk 3]
```diff
@@ -54,45 +59,13 @@ function resolveIdentifierToGraphQLCall(
...
-  while (ts.isIdentifier(element) && element !== prevElement) {
-    prevElement = element;
+  if (!checker) return null;
 
-    const definitions = info.languageService.getDefinitionAtPosition(
-      element.getSourceFile().fileName,
-      element.getStart()
-    );
+  const value = getValueOfIdentifier(input, checker);
+  if (!value) return null;
```
Comment (kitten) [inline packages/graphqlsp/src/ast/index.ts:65]: nit: redundant `null` conversions, which can be avoided if we adjust signatures here

File: packages/graphqlsp/src/persisted.ts [hunk 4]
```diff
@@ -248,37 +252,29 @@ export const getDocumentReferenceFromTypeQuery = (
...
-    if (found) return;
+  if (!typeChecker) return { node: null, filename };
+
+  // Handle EntityName (Identifier | QualifiedName)
+  let identifier: ts.Identifier | undefined;
+  if (ts.isIdentifier(typeQuery.exprName)) {
+    identifier = typeQuery.exprName;
+  } else if (ts.isQualifiedName(typeQuery.exprName)) {
+    // For qualified names like 'module.identifier', get the right-most identifier
+    identifier = typeQuery.exprName.right;
```
Comment (kitten) [inline packages/graphqlsp/src/persisted.ts:264]: nit: This seems a bit suspect. I don't remember fully what the intention here is, but I assume it's type casts? We may not be covering all edge cases here, if we do, this qualifies for making a helper and adding some comments

**Review by kitten — APPROVED** [review]
